tests/unit_tests/test_exceptions.py

```diff
@@ -0,0 +1,23 @@
+import pytest
+from bluesky.plan_stubs import null
+
+from hyperion.exceptions import WarningException, catch_exception_and_warn
+
+
+class TestException(Exception):
+    pass
+
+
+def dummy_plan():
+    raise TestException
+    yield from null
```

Comment:
Nit: If you swap the `yield from` and the raise then you can remove the `type: ignore` later on